reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/tests/FeatureScopeTest.kt

```diff
@@ -0,0 +1,206 @@
+class FeatureScopeTest : MockServerTest() {
+
+    @get:Rule
+    var forge = ForgeRule()
+        .useJvmFactories()
+        .useToolsFactories()
+        .withFactory(ConfigurationCoreForgeryFactory())
+
+    @StringForgery(type = StringForgeryType.ALPHABETICAL)
+    lateinit var fakeFeatureName: String
+
+    @Forgery
+    lateinit var fakeConfiguration: Configuration
+    private lateinit var fakeBatchData: List<RawBatchEvent>
+    private lateinit var fakeBatchMetadata: ByteArray
+    private lateinit var eventType: EventType
+    private lateinit var trackingConsent: TrackingConsent
+    private lateinit var stubFeature: Feature
+    private lateinit var testedInternalSdkCore: InternalSdkCore
+
+    @Before
+    fun setUp() {
+        fakeBatchData = forge.aList(size = forge.anInt(min = 1, max = 10)) {
+            val fakeEvent: JsonObject = forge.getForgery()
+            val eventMetadata = forge.anAlphabeticalString()
+            RawBatchEvent(
+                fakeEvent.toString().toByteArray(),
+                eventMetadata.toByteArray()
+            )
+        }
+        fakeBatchMetadata = forge.anAlphabeticalString().toByteArray()
+        eventType = forge.aValueFrom(EventType::class.java)
+        stubFeature = StubStorageBackedFeature(
+            forge,
+            fakeFeatureName,
+            getMockServerWrapper().getServerUrl()
+        )
+        fakeConfiguration = forge.getForgery()
+    }
+
+    @After
+    fun tearDown() {
+        cleanupStorage()
+        Datadog.stopInstance()
+        cleanMockWebServer()
+    }
+
+    // region Consent Granted
+
+    @Test
+    fun mustWriteTheEvents_whenFeatureWrite_trackingConsentGranted() {
+        // Given
+        trackingConsent = TrackingConsent.GRANTED
+        testedInternalSdkCore = Datadog.initialize(
+            context = ApplicationProvider.getApplicationContext(),
+            configuration = fakeConfiguration,
+            trackingConsent = trackingConsent
+        ) as InternalSdkCore
+        testedInternalSdkCore.registerFeature(stubFeature)
+        val featureScope = testedInternalSdkCore.getFeature(fakeFeatureName)
+
+        // When
+        checkNotNull(featureScope)
+        featureScope.withWriteContext { _, eventBatchWriter ->
+            fakeBatchData.forEach { rawBatchEvent ->
+                eventBatchWriter.write(
+                    rawBatchEvent,
+                    fakeBatchMetadata,
+                    eventType
+                )
+            }
+        }
+
+        // Then
+        ConditionWatcher {
+            MockWebServerAssert.assertThat(getMockServerWrapper())
+                .withConfiguration(fakeConfiguration)
+                .withTrackingConsent(trackingConsent)
+                .receivedData(fakeBatchData, fakeBatchMetadata)
+            true
+        }.doWait(LONG_WAIT_MS)
+    }
+
+    // endregion
+
+    // region Consent Pending
+
+    @Test
+    fun mustNotWriteTheEvents_whenFeatureWrite_trackingConsentPending() {
```

Comment:
If I understand correctly the name is more accurate as "mustNotReceiveTheEvents_..." than "mustNotWriteTheEvents_...", right?